### ISSUE TITLE
Runnable as new abstraction for function

### ISSUE THREAD

**Issue by dodokpeter** [spring-cloud/spring-cloud-function#121]
Is there a plan to add support for some other interfaces to spring cloud function, I see usefull especially interface for job execution - Runnable. 
Because some service can do only simple job without giving any feedback about result f.e.: data pump/sync, triggering reports. It is possible to use for this scenarios Consumer abstraction, but without input(it is not necessary), but with runnable it will be better understandable.

**Comment by dsyer** [comment]
Maybe `Supplier` would be better? You might want to return a status for the completed work or something, or a link to the result?

**Comment by dodokpeter** [comment]
Yes, you right, but sometimes response is not needed. My intension for Runnable is to use it for triggering some long running job. But I don't know if this usage  is ok for using cloud for function. Or f.e. there are some rest endpoints in our api, which definition correspond to Runnable - no input, no output. They only start job, f.e. checking some flag in all customers (and then send email them). At this point, we are not using cloud for function, but I started this question as possible enhancement.

**Comment by dsyer** [comment]
How do you trigger such tasks though? I would think there has to be an event to trigger it (even if it is just a cron or something). That event would naturally be the input to a `Consumer`, if there is no output, and a `Function` if there is. The more I think about it the more I expect `Function` to be used for 99% of use cases.

**Comment by dsyer** [comment]
FWIW I’m still on the fence on this, and while there are no specific features associated with “non-function” interfaces, the existing APIs and implementations do actually support them.
